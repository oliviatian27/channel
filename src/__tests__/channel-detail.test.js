import React from 'react';
import { shallow } from 'enzyme';

import ChannelDetail from '../components/channel-list/channel-detail';


describe('ChannelDetail tests', () => {

  it('renders ChannelDetail component', () => {
    const title = "apple";
    const description = "pear"
    const wrapper = shallow(<ChannelDetail title={title} description={description} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-detail')).toBeDefined();
    expect(wrapper.find('.channel-detail')).toMatchSnapshot();
  });

  it('renders ChannelTitle and ChannelDescription components and pass down the correct props', () => {
    const title = "apple";
    const description = "pear"

    const wrapper = shallow(<ChannelDetail title={title} description={description} />);

    // Check if the prop in the child Component is correct
    expect(wrapper.find('ChannelTitle').prop("title")).toEqual("apple");
    expect(wrapper.find('ChannelDescription').prop("description")).toEqual("pear");

    const newTitle = "peach";
    const newDescription = "grape"

   //update the props to check if the new prop passed down to the children component
   wrapper.setProps({ title: newTitle,description:newDescription });

   expect(wrapper.find('ChannelTitle').prop("title")).toEqual("peach");
   expect(wrapper.find('ChannelDescription').prop("description")).toEqual("grape");
  });

})
