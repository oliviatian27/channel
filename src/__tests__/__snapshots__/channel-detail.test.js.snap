// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelDetail tests renders ChannelDetail component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelDetail
      description="pear"
      title="apple"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ChannelTitle
            title="apple"
          />,
          <ChannelDescription
            description="pear"
          />,
        ],
        "className": "channel-detail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "apple",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "pear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ChannelTitle
              title="apple"
            />,
            <ChannelDescription
              description="pear"
            />,
          ],
          "className": "channel-detail",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "apple",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "pear",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ChannelTitle
          title="apple"
        />,
        <ChannelDescription
          description="pear"
        />,
      ],
      "className": "channel-detail",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "apple",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "pear",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ChannelTitle
            title="apple"
          />,
          <ChannelDescription
            description="pear"
          />,
        ],
        "className": "channel-detail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "apple",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "pear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ChannelTitle
            title="apple"
          />,
          <ChannelDescription
            description="pear"
          />,
        ],
        "className": "channel-detail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "apple",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "pear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
}
`;
